webknossos/webknossos/dataset/_utils/pims_images.py

```diff
@@ -300,12 +308,67 @@ def _normalize_original_images(self) -> Union[str, List[str]]:
         else:
             return str(original_images)
 
-    def _open_bioformats_images_raw(
+    def _ensure_correct_bioformats_usage(
+        self, images_context_manager: pims.FramesSequence
+    ) -> None:
+        if (
+            isinstance(images_context_manager, pims.bioformats.BioformatsReader)
+            and self._use_bioformats == False
+        ):  # None is allowed
+            raise RuntimeError(
+                "Selected bioformats reader, but using bioformats is not allowed "
+                + "(use_bioformats is False)."
+            )
+
+    def _try_open_pims_images(
+        self, original_images: Union[str, List[str]], exceptions: List[Exception]
+    ) -> Optional[pims.FramesSequence]:
+        if self._use_bioformats:
+            return None
+
+        open_kwargs = {}
+        if self._czi_channel is not None:
+            open_kwargs["czi_channel"] = self._czi_channel
+
+        def strategy_0() -> pims.FramesSequence:
+            result = pims.open(original_images, **open_kwargs)
+            self._ensure_correct_bioformats_usage(original_images)
+            return result
+
+        strategy_1 = lambda: pims.ImageSequence(original_images)
+
+        def strategy_2() -> pims.FramesSequence:
+            if isinstance(original_images, list):
+                # assuming the same reader works for all images:
+                first_image_handler = pims.open(original_images[0], **open_kwargs)
+                self._ensure_correct_bioformats_usage(first_image_handler)
+                return pims.ReaderSequence(
+                    original_images, type(first_image_handler), **open_kwargs
+                )
+            else:
+                return None
```

Comment:
This is new functionality, also the `use_bioformats` usage, everything else is just refactoring.